### PR TITLE
Add "Architecting a PWA with the App Shell Model"

### PR DESCRIPTION
**Pull request by addyosmani**

### CODE REVIEW

**Comment by PaulKinlan** [comment]
@petele Content looks good.  One question I have is if this should be under /web/progressive-web-apps/ or if we are cool with creating this under /web/fundamentals/ then we need to fix the overview url in the toc.

**Comment by petele** [comment]
@PaulKinlan - the plan is to move /web/pwa to /web/fund/pwa and really refresh that page. I'm not going to do it for the launch this morning, so will hold this for now. will get it live next week.

File: src/content/en/fundamentals/progressive-web-apps/app-shell.md [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+
+* HTML/CSS/JavaScript for the "skeleton" of your user interface
+* Navigation UI and logic
+* The code to display posts after they are retrieved from the server (and store
+  them locally using a storage mechanism like IndexedDB)
+
+Your `index.html` file in your work directory should look something like the
+following code. This is a subset of the actual contents and is not a complete
+index file:
+
```
Comment (jpmedley) [inline src/content/en/fundamentals/progressive-web-apps/app-shell.md]: As before, look at what I did in the edit.

File: src/content/en/fundamentals/progressive-web-apps/app-shell.md [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+### Benefits {: #app-shell-benefits }
+
+The benefits of an app shell architecture with a service worker include:
+
+* **Reliable performance that is consistently fast**. Repeat visits are
+extremely quick.  Static assets (e.g. HTML, JavaScript, images and CSS) are
+immediately cached locally so there is no need to re-fetch the shell (and
+optionally the content if that is cached too). The UI is cached locally and
+content is updated dynamically as required.
```
Comment (jpmedley) [inline src/content/en/fundamentals/progressive-web-apps/app-shell.md]: I forgot to change the default name of the commit. It's actually called 'Update app-shell.md.

Comment (addyosmani) [inline src/content/en/fundamentals/progressive-web-apps/app-shell.md]: Changes lgtm

File: src/content/en/fundamentals/progressive-web-apps/app-shell.md (hunk 1 above)
Comment (addyosmani) [inline src/content/en/fundamentals/progressive-web-apps/app-shell.md]: Changes LGTM

**Comment by addyosmani** [comment]
After talking to Pete, this might end up going into a new Architecture node.

**Pull request merged**